Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/internal/smooth_vertices.h

```diff
@@ -871,13 +871,9 @@ std::size_t smooth_vertices_on_surfaces(C3t3& c3t3,
           {
             const auto intersection = proj.value().first;
             using Segment = typename Tr::Geom_traits::Segment_3;
-            if (const Segment* s = std::get_if<Segment>(&intersection))
-              return CGAL::midpoint(s->source(), s->target());
-            else
-            {
-              CGAL_assertion(false);
-              return std::nullopt;
-            }
+            const Segment* s = std::get_if<Segment>(&intersection);
```

Comment:
it's not clear to me that here it can't happen. We are using a kernl with inexact constructions. @janetournois

Reply:
as the assertion still exist and nullptr output is not handled in the calling code, it's not a regression.